src/storage/mvcc/reader/forward_seeker.rs

```diff
@@ -0,0 +1,296 @@
+    /// TODO: Is the range `[lower_bound, upper_bound)`?
+    ///
+    /// Default is `(None, None)`.
+    #[inline]
+    pub fn range(mut self, lower_bound: Option<Vec<u8>>, upper_bound: Option<Vec<u8>>) -> Self {
+        self.lower_bound = lower_bound;
+        self.upper_bound = upper_bound;
+        self
+    }
+
+    /// Build `ForwardSeeker` from the current configuration.
+    pub fn build(self) -> Result<ForwardSeeker<S>> {
+        let lock_cursor = CursorBuilder::new(&self.snapshot, CF_LOCK)
+            .bound(self.lower_bound.clone(), self.upper_bound.clone())
+            .fill_cache(self.fill_cache)
+            .build()?;
+
+        let write_cursor = CursorBuilder::new(&self.snapshot, CF_WRITE)
+            .bound(self.lower_bound.clone(), self.upper_bound.clone())
+            .fill_cache(self.fill_cache)
+            .build()?;
+
+        Ok(ForwardSeeker {
+            snapshot: self.snapshot,
+            fill_cache: self.fill_cache,
+            omit_value: self.omit_value,
+            isolation_level: self.isolation_level,
+            lower_bound: self.lower_bound,
+            upper_bound: self.upper_bound,
+            lock_cursor,
+            write_cursor,
+            default_cursor: None,
+            statistics: Statistics::default(),
+        })
+    }
+}
+
+/// This struct can be used to find next key greater or equal to a given user key. Internally,
+/// rollbacks are ignored and smaller version will be tried. If the isolation level is SI, locks
+/// will be checked first.
+///
+/// This struct keeps the iterator moves forward. If you use this to perform seeking multiple times,
+/// then you are not allowed to seek a key that is less than the previous one.
+///
+/// Use `ForwardSeekerBuilder` to build `ForwardSeeker`.
+pub struct ForwardSeeker<S: Snapshot> {
+    snapshot: S,
+    fill_cache: bool,
+    omit_value: bool,
+    isolation_level: IsolationLevel,
+
+    /// `lower_bound` and `upper_bound` is only used to create `default_cursor`. It will be consumed
+    /// after default_cursor's being created.
+    lower_bound: Option<Vec<u8>>,
+    upper_bound: Option<Vec<u8>>,
+
+    lock_cursor: Cursor<S::Iter>,
+    write_cursor: Cursor<S::Iter>,
+
+    /// `default cursor` is lazy created only when it's needed.
+    default_cursor: Option<Cursor<S::Iter>>,
+
+    statistics: Statistics,
+}
+
+impl<S: Snapshot> ForwardSeeker<S> {
+    /// Take out and reset the statistics collected so far.
+    pub fn take_statistics(&mut self) -> Statistics {
+        ::std::mem::replace(&mut self.statistics, Statistics::default())
+    }
+
+    /// Get reference of the statics collected so far.
+    pub fn get_statistics(&self) -> &Statistics {
+        &self.statistics
+    }
+
+    /// Get the next key-value pair, where the key is greater or equal to given `key`.
+    pub fn read_next(&mut self, mut key: Key, ts: u64) -> Result<Option<(Key, Value)>> {
```

Comment:
Are we going to switch the signature to `read_next()` (i.e. use next internally) instead of `read_next(key, ts)` (i.e. use prefix_next to make the next key) in this PR or in future PRs?  @zhangjinpeng1987

Reply:
Remove `key` parameter for this function.